src/formatDate.js

```diff
@@ -9,6 +9,41 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  const oldSeparator = fromFormat.pop();
+  const newSeparator = toFormat.pop();
+
+  const splitedDate = date.split(oldSeparator);
+
+  const newDate = [];
+
+  let yearFormat = toFormat.includes('YY') ? 'YY' : 'YYYY';
```

Comment:
The variable `yearFormat` is being determined based on the presence of 'YY' in the `toFormat` array. However, this might not work as expected if the `fromFormat` contains 'YYYY' and the `toFormat` contains 'YY', or vice versa. This logic needs to be more robust to handle the conversion between year formats correctly.